buttplug/src/device/protocol/generic_command_manager.rs

```diff
@@ -172,6 +172,15 @@ impl GenericCommandManager {
     }
   }
 
+  pub fn get_vibration(&mut self) -> Vec<Option<u32>> {
```

Comment:
This can be simplified down to single line:

```rust
self.vibrations.iter().map(|x| Some(x)).collect()
```
I'll handle it in a followup thought.